### PR TITLE
Changed code so that it will be compatible with c++.

### PR DESCRIPTION
**Pull request by shmuelnatan**
Added #ifdef extern "C" to public header files.
Added (char*) casting to const strings.

### CODE REVIEW

File: cligen_file.c [hunk 1]
```diff
@@ -153,16 +153,16 @@ cli_expand_cb(cligen_handle h,
 #if 1
     /* Special case for two partly overlapping expand sets */
     if (strcmp(fn_str,"exp")==0){
-	cvec_add_string(commands, NULL, "exp1"); cvec_add_string(helptexts, NULL, "Help exp1");
-	cvec_add_string(commands, NULL, "exp2"); cvec_add_string(helptexts, NULL, "Help exp2");
-	cvec_add_string(commands, NULL, "exp3"); cvec_add_string(helptexts, NULL, "Help exp3");
+	cvec_add_string(commands, NULL, (char*)"exp1"); cvec_add_string(helptexts, NULL, (char*)"Help exp1");
```
Comment (shmuelhazan) [inline cligen_file.c:156]: What is this change for?

File: cligen_file.c [hunk 2]
```diff
@@ -153,16 +153,16 @@ cli_expand_cb(cligen_handle h,
...
     }
     else{
-	cvec_add_string(commands, NULL, "exp2");  cvec_add_string(helptexts, NULL, "Help exp2");
+	cvec_add_string(commands, NULL, (char*)"exp2");  cvec_add_string(helptexts, NULL, (char*)"Help exp2");
     }
 #else
-    cvec_add_string(commands, NULL, "auto");
-    cvec_add_string(helptexts, NULL, "This expansion is auto-generated by -e option");
+    cvec_add_string(commands, NULL, (char*)"auto");
+    cvec_add_string(helptexts, NULL, (char*)"This expansion is auto-generated by -e option");
```
Comment (shmuelhazan) [inline cligen_file.c:165]: Same. I don't understand. This is a C file. Why do we have to modify it?

File: cligen_hello.c [hunk 3]
```diff
@@ -63,7 +63,7 @@ str2fn(char *name, void *arg, char **error)
 
 
 /*! The command syntax specification */
-static char *hello_syntax = "prompt=\"hello> \";\n" 
+static char *hello_syntax = (char*)"prompt=\"hello> \";\n"
```
Comment (shmuelhazan) [inline cligen_hello.c:66]: Again. C file.

File: cligen.h [hunk 4]
```diff
@@ -38,6 +38,10 @@
    You include cligen.h which includes the other cligen include files for you.
```
Comment (shmuelhazan) [inline cligen.h:38]: These changes are enough. All other changes are simply not needed. I don't understand what is the reason for them.

**Comment by olofhagsand** [comment]
I agree with the above comments from @shmuelhazan , only the .h file change should be necessary.
Also, cligen_file/cligen_tutorial/cligen_hello are just applications, they are not part of the lib exported to clixon.

**Pull request closed without merging**